components/script/dom/urlhelper.rs

```diff
@@ -92,4 +92,18 @@ impl UrlHelper {
             let _ = quirks::set_username(url, &value.0);
         }
     }
+    // https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy
+    pub fn is_origin_trustworthy(url: &ServoUrl) -> bool {
+        // Step 3
+        if url.scheme() == "http" || url.scheme() == "wss" {
```

Comment:
Yes, except that we don't support HTTPS in our automated tests yet. Certainly a FIXME comment here would be an improvement.